# Identifying Syntretus infections
Several thousands of insects are sampled from the wild each year to have their genomes sequences. Often times, the researchers producing this sequence data will also unknowingly produce sequence data of host-infecting passengers (e.g. viruses, microbial symbionts and parasites). These unintended (but fortunate) outcomes can be a treasure trove for researchers who study these insects as hosts as it can provide not only sequencing data of the passenger, but also expand our understanding of host range and geographic range of the passenger. For the purposes of this study, we sought to identify _Syntretus_ [Braconidae:Euphorinae] wasp infections in publically available read data two know potential hosts including _Drosophila_ and _Apis_ _cerena_. Due to the sheer number of SRA files we needed to explore and their large file sizes, it was inefficient for us to do individual blast searches of each individual SRA file. Instead, we developed a bash 'for loop' that takes a .txt list of SRA accessions that we are interested in and downloads each one at a time to be screened for _Syntretus_ markers. Specifically, the code uses the fasterq-dump function of ncbi-sra/3.0.2 to upload the next SRA accession in the list. The uploaded SRA read file is then submitted to BBmap/38.35 software where the reads are mapped against both the host mitochondrial genome and the _Syntretus_ mitochondrial genome. The outputs of BBmap/38.35 are coverage statistics and mapped reads. Once the mapping is finished, the SRA read file is deleted to make room for the next. As the 'for loop' implies, this continues up until the last SRA accession in the list. 
